src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -0,0 +1,78 @@
+package mate.academy.dao.impl;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.MovieSessionDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.MovieSession;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.persist(movieSession);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add movie session to DB" + movieSession,
+                    e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return movieSession;
+    }
+
+    @Override
+    public Optional<MovieSession> get(Long id) {
+        String hql = "from MovieSession ms "
+                + "left join fetch ms.movie "
+                + "left join fetch ms.cinemaHall "
+                + "where ms.id = :id";
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> movieSessionQuery = session.createQuery(hql, MovieSession.class);
```

Comment:
can we just `return Optional.ofNullable(session.get(MovieSession.class, id));` ?

Reply:
Yes, we can, but the checklist has three methods. I thought I needed to choose one of them.
https://github.com/vfilimo/hibernate-create-movie-session-hw/blob/master/checklist.md#:~:text=Think%20what%20the%20difference%20between%20getSingleResult()%2C%20uniqueResult()%20and%20uniqueResultOptional()%3F%20Which%20one%20should%20we%20use%3F

Reply:
No, because the Query class does not have a get() method.

Reply:
but why do you need query? 
Make these fields not lazy, cause you have ManyToOne  relation, not OneToMany, and then just use code suggested by Olena